ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/QueryAnonymizer.java

```diff
@@ -887,15 +887,31 @@ private static String visitWithinExpression(final WithinExpressionContext contex
         final String after =
             getSizeAndUnitFromJoinWindowSize(beforeAndAfterJoinWindow.joinWindowSize(1));
         stringBuilder.append(String.format("(%s, %s)", before, after));
+      } else if (context instanceof SqlBaseParser.JoinWindowWithGraceContext) {
+        final SqlBaseParser.JoinWindowWithGraceContext joinWithGracePeriod =
+            (SqlBaseParser.JoinWindowWithGraceContext) context;
+
+        stringBuilder.append(anonymizeJoinWindowWithGrace(joinWithGracePeriod));
       } else {
         throw new RuntimeException("Expecting either a single join window, ie \"WITHIN 10 "
-            + "seconds\", or a join window with before and after specified, "
-            + "ie. \"WITHIN (10 seconds, 20 seconds)");
+            + "seconds\", a join window with before and after specified, "
+            + "ie. \"WITHIN (10 seconds, 20 seconds)\", or a join window with a grace period, "
+            + "ie. \"WITHIN (SIZE 10 seconds, GRACE PERIOD 5 seconds)\"");
       }
 
       return stringBuilder.toString();
     }
 
+    private static String anonymizeJoinWindowWithGrace(
```

Comment:
Should we unify this method with the existing `getSizeAndUnitFromJoinWindowSize` ?

Reply:
I made some changes on this method.